src/Chapter3.hs

```diff
@@ -972,6 +1042,16 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay = Monday | Tuesday | Wensday | Thusday | Friday | Saturday | Sunday deriving (Show, Eq, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend d = (d == Saturday) || (d == Sunday)
+
+nextDay :: WeekDay -> WeekDay
+nextDay = succ
```

Comment:
Unfortunately, `succ` would fail on the edge case of the enum (in this case on the `Sunday`. To avoid it, you can pattern match on this specific case like:

```haskell
nextDay Sunday = Monday
nextDay d = succ d
```

You could also use the instance of `Bounded` here:

```haskell
nextDay day
    | day == maxBound = minBound
    | otherwise = succ day
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂